packages/filecoin-api/src/aggregator/buffer-reducing.js

```diff
@@ -182,7 +187,7 @@ export function aggregatePieces(bufferedPieces, sizes) {
   /** @type {BufferedPiece[]} */
   const remainingBufferedPieces = []
 
-  for (const bufferedPiece of bufferedPieces) {
+  for (const bufferedPiece of transformedBufferedPieces) {
     const p = Piece.fromLink(bufferedPiece.piece)
     if (builder.estimate(p).error) {
       remainingBufferedPieces.push(bufferedPiece)
```

Comment:
If you somehow configure big piece(s) to prepend that exceed `maxAggregateSize`, then `remainingBufferedPieces` might end up with a configured prepend piece 😱 .

Perhaps we should not concat `prependBufferedPieces` with `bufferedPieces`, and instead iterate over them first adding them to `addedBufferedPieces` _before_ iterating over `bufferedPieces`, and just throw if `estimate` errors since we cannot build an aggregate consisting of _only_ prepended pieces.

Reply:
ok, that seems reasonable considering other could implement this.